optimade/filtertransformers/mongo.py

```diff
@@ -216,11 +218,11 @@ def set_zip_op_rhs(self, arg):
         # ANY value_zip_list )
         raise NotImplementedError
 
-    def predicate_comparison(self, arg):
-        # predicate_comparison: LENGTH property OPERATOR value
+    def length_op_rhs(self, arg):
+        # length_op_rhs: LENGTH [ OPERATOR ] value
         # TODO: https://stackoverflow.com/questions/7811163/query-for-documents-where-array-size-is-greater-than-1
-        if arg[2] == "=":
-            return {arg[1]: {"$size": arg[3]}}
+        if len(arg) == 2 or (len(arg) == 3 and arg[1] == "="):
+            return {"$size": arg[-1]}
 
         raise NotImplementedError
```

Comment:
Should we add the slow `$where` search here as a fallback (from @fekad's suggestions)?

If not, making the error message more informative would be good. My suggestion below only works if arg is only ever length 2 or 3, which I think is the case (without testing)

```suggestion
raise NotImplementedError(f"Operator {arg[1]} not implemented for LENGTH filter.")
```

Reply:
I don't know `mongo` at all, so whatever you think or know to be right, you just add :)

Reply:
And `arg` would indeed never be any other length than 2 or 3.